### PR TITLE
Solve review comment for Azure.ResourceManager.AppService

### PR DESCRIPTION
**Pull request by ArthurMa1978**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 1]
```diff
@@ -4453,6 +4415,111 @@ public partial class AppRegistration
...
+    {
+        internal AppServiceCertificateDetails() { }
+        public string Issuer { get { throw null; } }
+        public System.DateTimeOffset? NotAfter { get { throw null; } }
+        public System.DateTimeOffset? NotBefore { get { throw null; } }
+        public string RawData { get { throw null; } }
+        public string SerialNumber { get { throw null; } }
+        public string SignatureAlgorithm { get { throw null; } }
+        public string Subject { get { throw null; } }
+        public string Thumbprint { get { throw null; } }
```
Comment (Yao725) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4513]: BinaryData?

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 2]
```diff
@@ -7524,6 +7491,39 @@ public partial class VnetValidationTestFailure : Azure.ResourceManager.Models.Re
...
+        public bool? IsEnabled { get { throw null; } set { } }
+        public string Kind { get { throw null; } set { } }
+        public System.Uri StorageAccountUri { get { throw null; } set { } }
+    }
+    public partial class WebAppBackupSchedule
+    {
+        public WebAppBackupSchedule(int frequencyInterval, Azure.ResourceManager.AppService.Models.FrequencyUnit frequencyUnit, bool shouldKeepAtLeastOneBackup, int retentionPeriodInDays) { }
+        public int FrequencyInterval { get { throw null; } set { } }
+        public Azure.ResourceManager.AppService.Models.FrequencyUnit FrequencyUnit { get { throw null; } set { } }
+        public System.DateTimeOffset? LastExecutionOn { get { throw null; } }
```
Comment (Yao725) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:7509]: LastExecutedOn?

**Review by Yao725 — APPROVED** [review]
FrontDoor LGTM, left a few comments about AppService

File: sdk/websites/Azure.ResourceManager.AppService/src/autorest.md [hunk 3]
```diff
@@ -105,9 +105,22 @@ override-operation-name:
   Domains_ListRecommendations: GetAppServiceDomainRecommendations
   Domains_CheckAvailability: CheckAppServiceDomainRegistrationAvailability
   Recommendations_DisableRecommendationForSubscription: DisableAppServiceRecommendation
-  ResourceHealthMetadata_ListByResourceGroup: GetAllResourceHealthMetadata
   WebApps_ListSnapshotsSlot: GetSlotSnapshots
   WebApps_ListSnapshotsFromDRSecondarySlot: GetSlotSnapshotsFromDRSecondary
+  # All bellowing operations should be EBNerver once the polymorphic change is ready
+  AppServiceEnvironments_ListWebApps: GetAllWebAppData
+  ResourceHealthMetadata_ListByResourceGroup: GetAllResourceHealthMetadataData
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/src/autorest.md:112]: There are case use this kind of naming, `result` is not consistency with other operations, so keep it as is

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 4]
```diff
@@ -4453,6 +4415,111 @@ public partial class AppRegistration
...
+    public partial class AppServiceAadAllowedPrincipals
+    {
+        public AppServiceAadAllowedPrincipals() { }
+        public System.Collections.Generic.IList<string> Groups { get { throw null; } }
+        public System.Collections.Generic.IList<string> Identities { get { throw null; } }
+    }
+    public partial class AppServiceAadLoginFlow
+    {
+        public AppServiceAadLoginFlow() { }
+        public bool? IsDisableWwwAuthenticate { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4427]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4513]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:7509]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 5]
```diff
@@ -4453,6 +4415,111 @@ public partial class AppRegistration
...
+        public string Name { get { throw null; } }
+        public string Product { get { throw null; } }
+        public string PromotionCode { get { throw null; } }
+        public string Publisher { get { throw null; } }
+        public string Version { get { throw null; } }
+    }
+    public partial class AppServiceBillingMeter : Azure.ResourceManager.Models.ResourceData
+    {
+        public AppServiceBillingMeter() { }
+        public string BillingLocation { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4481]: Azure Location?

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 6]
```diff
@@ -4453,6 +4415,111 @@ public partial class AppRegistration
...
+        public string Kind { get { throw null; } set { } }
+        public System.Guid? MeterId { get { throw null; } set { } }
+        public double? Multiplier { get { throw null; } set { } }
+        public string OSType { get { throw null; } set { } }
+        public string ShortName { get { throw null; } set { } }
+    }
+    public partial class AppServiceBlobStorageApplicationLogsConfig
+    {
+        public AppServiceBlobStorageApplicationLogsConfig() { }
+        public Azure.ResourceManager.AppService.Models.LogLevel? Level { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4492]: Model name LogLevel is too generic.

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 7]
```diff
@@ -4528,30 +4635,82 @@ public partial class AppServiceDomainPatch : Azure.ResourceManager.Models.Resour
...
+        public int? Port { get { throw null; } }
+    }
+    public partial class AppServiceEnvironmentAddressResult : Azure.ResourceManager.Models.ResourceData
+    {
+        public AppServiceEnvironmentAddressResult() { }
+        public System.Net.IPAddress InternalIPAddress { get { throw null; } set { } }
+        public string Kind { get { throw null; } set { } }
+        public System.Collections.Generic.IList<System.Net.IPAddress> OutboundIPAddresses { get { throw null; } }
+        public System.Net.IPAddress ServiceIPAddress { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.AppService.Models.VirtualIPMapping> VipMappings { get { throw null; } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4708]: The property name prefix `Vip` is misleading, it should be `VIPMappings` or consider expand it to `VirtualIPMappings`.

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 8]
```diff
@@ -4572,29 +4731,151 @@ public partial class AppServiceEnvironmentPatch : Azure.ResourceManager.Models.R
...
+        InProgress = 0,
+        Failed = 1,
+        Succeeded = 2,
+        TimedOut = 3,
+        Created = 4,
+    }
+    public partial class AppServicePlanPatch : Azure.ResourceManager.Models.ResourceData
+    {
+        public AppServicePlanPatch() { }
+        public System.DateTimeOffset? FreeOfferExpirationOn { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4876]: FreeOfferExpireOn

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 9]
```diff
@@ -4774,6 +5115,25 @@ public partial class AppServiceValidateResult
...
+        public bool? IsSwift { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.AppService.Models.AppServiceVnetRoute> Routes { get { throw null; } }
+        public Azure.Core.ResourceIdentifier VnetResourceId { get { throw null; } }
+    }
+    public partial class AppServiceVnetRoute : Azure.ResourceManager.Models.ResourceData
+    {
+        public AppServiceVnetRoute() { }
+        public string EndAddress { get { throw null; } set { } }
+        public string Kind { get { throw null; } set { } }
+        public Azure.ResourceManager.AppService.Models.RouteType? RouteType { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:5134]: Model name `RouteType` is too generic.

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 10]
```diff
@@ -3138,14 +3105,47 @@ public partial class VnetInfoResourceData : Azure.ResourceManager.Models.Resourc
...
-        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.AppService.Models.VnetRoute> Routes { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.AppService.Models.AppServiceVnetRoute> Routes { get { throw null; } }
         public Azure.Core.ResourceIdentifier VnetResourceId { get { throw null; } set { } }
     }
+    public partial class WebAppBackupData : Azure.ResourceManager.Models.ResourceData
+    {
+        public WebAppBackupData() { }
+        public int? BackupId { get { throw null; } }
+        public string BlobName { get { throw null; } }
+        public string CorrelationId { get { throw null; } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:3121]: I see `CorrelationId` in another place is `Guid`. This should probably also be `Guid`?

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 11]
```diff
@@ -5165,12 +5339,12 @@ public partial class CloningInfo
...
         public System.Collections.Generic.IDictionary<string, string> AppSettingsOverrides { get { throw null; } }
+        public bool? CanOverwrite { get { throw null; } set { } }
         public bool? CloneCustomHostNames { get { throw null; } set { } }
         public bool? CloneSourceControl { get { throw null; } set { } }
         public bool? ConfigureLoadBalancing { get { throw null; } set { } }
         public System.Guid? CorrelationId { get { throw null; } set { } }
         public string HostingEnvironment { get { throw null; } set { } }
-        public bool? Overwrite { get { throw null; } set { } }
         public Azure.Core.ResourceIdentifier SourceWebAppId { get { throw null; } set { } }
         public string SourceWebAppLocation { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:5349]: AzureLocation?

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 12]
```diff
@@ -7538,6 +7491,53 @@ public partial class WebAppMinorVersion
         public Azure.ResourceManager.AppService.Models.WebAppRuntimes StackSettings { get { throw null; } }
         public string Value { get { throw null; } }
     }
+    public partial class WebAppMSDeploy : Azure.ResourceManager.Models.ResourceData
+    {
+        public WebAppMSDeploy() { }
+        public string ConnectionString { get { throw null; } set { } }
+        public string DbType { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:7498]: DBType

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 8 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4876]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs [hunk 13]
```diff
@@ -3138,14 +3105,47 @@ public partial class VnetInfoResourceData : Azure.ResourceManager.Models.Resourc
...
+        public string BlobName { get { throw null; } }
+        public string CorrelationId { get { throw null; } }
+        public System.DateTimeOffset? CreatedOn { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.AppService.Models.AppServiceDatabaseBackupSetting> Databases { get { throw null; } }
+        public System.DateTimeOffset? FinishedOn { get { throw null; } }
+        public bool? IsScheduled { get { throw null; } }
+        public string Kind { get { throw null; } set { } }
+        public System.DateTimeOffset? LastRestoreOn { get { throw null; } }
+        public string Log { get { throw null; } }
+        public string NamePropertiesName { get { throw null; } }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:3129]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 5 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4481]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4492]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 7 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:4708]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 9 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:5134]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 11 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:5349]: fixed

File: sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs (hunk 12 above)
Comment (ArthurMa1978) [reply sdk/websites/Azure.ResourceManager.AppService/api/Azure.ResourceManager.AppService.netstandard2.0.cs:7498]: fixed

**Pull request merged**